lib/origen_swd/driver.rb

```diff
@@ -177,11 +177,28 @@ def send_header(apndp, rnw, address)
     end
 
     # Waits appropriate number of cycles for the acknowledgement phase
-    def receive_acknowledgement
+    # @param options [Hash] Options during acknowledgement reception.
+    # @option options [true, false] :ignore_acknowledgement Indicates that the acknowledgement
+    #   should not be checked for a pass/fail result. Rather, the tester will just be cycled 3 times.
+    # @option options [true, false] :expect_transaction_error Instead of verifying for a successful
+    #   transaction, verifies that a transaction error occurred.
+    # @note If both :ignore_acknowledgement and :expect_transaction_error are present,
+    #   :ignore_acknowledgement will take precedence.
+    def receive_acknowledgement(options = {})
```

Comment:
Should you also have default settings for the various options here?  I know you check all cases below but a good practice.

Reply:
The default will be `nil`, so they'll all be skipped by `Hash` access definition; essentially `false`. And the defaults would only be used internally, so I'm not sure we gain anything.

Reply:
Unless you're talking about the YARDOC specifically?